File: server-java/src/main/java/com/parentingapp/server/auth/AuthController.java
package com.parentingapp.server.auth;

import com.parentingapp.server.auth.dto.AuthResponse;
import com.parentingapp.server.auth.dto.GoogleLoginRequest;
import com.parentingapp.server.auth.dto.KakaoLoginRequest;
import com.parentingapp.server.auth.dto.LoginRequest;
import com.parentingapp.server.auth.dto.RefreshRequest;
import com.parentingapp.server.auth.dto.SignupRequest;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final AuthService authService;

    public AuthController(AuthService authService) {
        this.authService = authService;
    }

    @PostMapping("/signup")
    public ResponseEntity<AuthResponse> signup(@Valid @RequestBody SignupRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(authService.signup(request));
    }

    @PostMapping("/login")
    public AuthResponse login(@Valid @RequestBody LoginRequest request) {
        return authService.login(request);
    }

    @PostMapping("/refresh")
    public AuthResponse refresh(@Valid @RequestBody RefreshRequest request) {
        return authService.refresh(request);
    }

    @PostMapping("/logout")
    public ResponseEntity<Void> logout(@Valid @RequestBody RefreshRequest request) {
        authService.logout(request);
        return ResponseEntity.noContent().build();
    }

    @PostMapping("/kakao")
    public AuthResponse kakaoLogin(@Valid @RequestBody KakaoLoginRequest request) {
        return authService.loginWithKakao(request);
    }

    @PostMapping("/google")
    public AuthResponse googleLogin(@Valid @RequestBody GoogleLoginRequest request) {
        return authService.loginWithGoogle(request);
    }
}

File: server/src/modules/auth/auth.controller.ts
import type { Request, Response } from "express";
import * as authService from "./auth.service";

export async function signupHandler(req: Request, res: Response) {
  const result = await authService.signup(req.body);
  res.status(201).json(result);
}

export async function loginHandler(req: Request, res: Response) {
  const result = await authService.login(req.body);
  res.status(200).json(result);
}

export async function refreshHandler(req: Request, res: Response) {
  const result = await authService.refresh(req.body);
  res.status(200).json(result);
}

export async function logoutHandler(req: Request, res: Response) {
  await authService.logout(req.body);
  res.status(204).send();
}

export async function kakaoLoginHandler(req: Request, res: Response) {
  const result = await authService.loginWithKakao(req.body);
  res.status(200).json(result);
}

export async function googleLoginHandler(req: Request, res: Response) {
  const result = await authService.loginWithGoogle(req.body);
  res.status(200).json(result);
}
